### PR TITLE
on 32bit platform diskq ftruncate could fail due to size 32/64 interface

### PR DESCRIPTION
**Pull request by Kokan**
On 32bit platforms glong is 32bit, therefore we explicitly casted
a signed 64-bit value to a signed 32-bit value, which caused an EINVAL error
on passing a negative value to ftruncate.

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: modules/diskq/qdisk.c [hunk 1]
```diff
@@ -437,7 +437,8 @@ qdisk_pop_head(QDisk *self, GString *record)
         {
           self->hdr->backlog_head = self->hdr->read_head;
 
-          if (!self->options->read_only && self->hdr->length == 0)
+          g_assert(self->hdr->backlog_len == 0);
```
Comment (MrAnno) [inline modules/diskq/qdisk.c]: @gaborznagy I think this deserves a separate commit with some explanation.
What do you think?

File: modules/diskq/qdisk.c [hunk 2]
```diff
@@ -216,7 +224,7 @@ _truncate_file(QDisk *self, gint64 new_size)
       msg_error("Error truncating disk-queue file",
                 evt_tag_error("error"),
                 evt_tag_str("filename", self->filename),
-                evt_tag_long("expected-size", new_size),
+                evt_tag_printf("expected-size", "%ld", new_size),
```
[the remark this answers is not in the document]
Comment (MrAnno) [reply modules/diskq/qdisk.c]: `evt_tag_long` already uses `long long`. Why do we need this modification?

File: modules/diskq/qdisk.c (hunk 1 above)
Comment (MrAnno) [reply modules/diskq/qdisk.c]: Discussed IRL, this belongs to the patch.

**Comment by kira-syslogng** [comment]
Build SUCCESS
